Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBox.cs

```diff
@@ -31,16 +30,69 @@ public enum FlattenModeEnum
             FlattenAuto,    // Flatten the smallest relative axis if it falls below threshold
         }
 
-        #region public
+        [Header("Objects")]
+        [Tooltip("The target object")]
+        [SerializeField]
+        protected GameObject target;
+
+        [Tooltip("The transform used to scale the bounding box (will be auto-generated)")]
+        [SerializeField]
+        protected Transform scaleTransform = null;
+
+        [Header("Flattening & Padding")]
+        [Tooltip("Flattening behavior setting.")]
+        [SerializeField]
+        protected FlattenModeEnum flattenPreference = FlattenModeEnum.FlattenAuto;
+
+        [Tooltip("The relative % size of an axis must meet before being auto-flattened")]
+        [SerializeField]
+        protected float flattenAxisThreshold = 0.025f;
+
+        [Tooltip("The relative % size of a flattened axis")]
+        [SerializeField]
+        protected float flattenedAxisThickness = 0.01f;
+
+        [Tooltip("How much to pad the scale of the box to fit around objects (as % of largest dimension)")]
+        [SerializeField]
+        protected float scalePadding = 0.05f;
+
+        [Tooltip("How much to pad the scale of the box on an axis that's flattened")]
+        [SerializeField]
+        protected float flattenedScalePadding = 0f;
+
+        [Header("Bounds Calculation")]
+        [Tooltip("Method used to calculate the bounds of the object.")]
+        [SerializeField]
+        protected BoundsCalculationMethodEnum boundsCalculationMethod = BoundsCalculationMethodEnum.MeshFilterBounds;
+
+        [Tooltip("Any renderers on this layer will be ignored when calculating object bounds")]
+        [SerializeField]
+        protected LayerMask ignoreLayers = (1 << 2); // Ignore Raycast Layer
+
+        protected Vector3 targetBoundsWorldCenter = Vector3.zero;
+
+        protected Vector3 targetBoundsLocalScale = Vector3.zero;
+
+        protected Bounds localTargetBounds = new Bounds();
+
+        protected List<Vector3> boundsPoints = new List<Vector3>();
+
+        protected FlattenModeEnum flattenedAxis = FlattenModeEnum.DoNotFlatten;
+
+        protected bool isVisible = true;
+
+        protected Renderer rendererForVisibility;
 
         public Action OnFlattenedAxisChange;
 
-        public virtual FlattenModeEnum FlattenPreference {
+        public virtual FlattenModeEnum FlattenPreference
```

Comment:
Please co-locate these properties with the serialized field for readability / discoverability.